web/src/settings_linkifiers.ts

```diff
@@ -219,20 +241,21 @@ export function build_page() {
             $linkifier_status.hide();
             $pattern_status.hide();
             $template_status.hide();
-            const linkifier = {};
+            const linkifier: {[key in string]?: string} & {id?: number} = {};
 
             for (const obj of $(this).serializeArray()) {
                 linkifier[obj.name] = obj.value;
             }
 
-            channel.post({
+            void channel.post({
                 url: "/json/realm/filters",
                 data: $(this).serialize(),
                 success(data) {
+                    const clean_data = configure_linkifier_api_response_schema.parse(data);
                     $("#linkifier_pattern").val("");
                     $("#linkifier_template").val("");
                     $add_linkifier_button.prop("disabled", false);
-                    linkifier.id = data.id;
+                    linkifier.id = clean_data.id;
                     ui_report.success(
                         $t_html({defaultMessage: "Custom linkifier added!"}),
```

Comment:
Not related to this PR or migration work, but do we want this `ui_report`/`defaultMessage` to be sticky. Cause similar features like `settings_playground` provide `remove_after` field of 3000s in the [success](https://github.com/zulip/zulip/blob/e7a19aca757820fe53f896ab15ad4128bf94838a/web/src/settings_playgrounds.ts#L132) `ui_report`.

Reply:
Likely we want this to match the pattern of others, yeah.